iidm/iidm-reducer/src/main/java/com/powsybl/iidm/reducer/DefaultNetworkReducer.java

```diff
@@ -96,20 +96,24 @@ protected void reduce(ThreeWindingsTransformer transformer) {
         } else {
             transformer.remove();
         }
-
         observers.forEach(o -> o.transformerRemoved(transformer));
     }
 
     @Override
     protected void reduce(HvdcLine hvdcLine) {
-        VoltageLevel vl1 = hvdcLine.getConverterStation1().getTerminal().getVoltageLevel();
-        VoltageLevel vl2 = hvdcLine.getConverterStation2().getTerminal().getVoltageLevel();
-        if (getPredicate().test(vl1) || getPredicate().test(vl2)) {
-            throw new UnsupportedOperationException("Reduction of HVDC lines is not supported");
+        Terminal terminal1 = hvdcLine.getConverterStation1().getTerminal();
+        Terminal terminal2 = hvdcLine.getConverterStation2().getTerminal();
+        VoltageLevel vl1 = terminal1.getVoltageLevel();
+        VoltageLevel vl2 = terminal2.getVoltageLevel();
+        HvdcConverterStation<?> station1 = hvdcLine.getConverterStation1();
+        HvdcConverterStation<?> station2 = hvdcLine.getConverterStation2();
+        if (getPredicate().test(vl1)) {
```

Comment:
We should test first if vl1 AND vl1 have to be removed ?

Reply:
My question is just: does it work if both voltage levels have to be removed?

Reply:
If both voltage levels have to be removed, test(vl1) = test(vl2) = false so we remove the hvdcLine (l.115).
However I have to add here the removal of converter stations as well.

Reply:
maybe in that case, it is automatically removed with the voltage level?

Reply:
In fact the last comment is not necessary : converter stations are in this case already removed when removing the voltage levels.